gufe/chemicalstate.py

```diff
@@ -101,9 +101,9 @@ def identifier(self):
         return self._identifier
 
     @property
-    def charge(self):
-        """Total charge for the ChemicalState."""
-        return sum([component.charge for component in self._components.values()])
+    def formal_charge(self):
+        """Formal charge for the ChemicalState."""
+        return sum([component.formal_charge for component in self._components.values()])
```

Comment:
I think this needs to check if there's a neutralizing solvent, if so return 0 else return the sum ignoring nans.  So...

```python
if any(hasattr('neutralize', comp) and comp.neutralize
         for comp in self._components.values()):
    return 0
```

Which is a tad ugly tbh, but not sure how else to work around solvents.

Reply:
What happens here if a box of water & ions are passed through? (say a membrane protein where you need to define the water component because otherwise most solvation routines are too dumb and might add water inside the membrane?)

Reply:
Hmm or it could be that solvents look neutral and the edge gets to decide their actual charge? Like how box info is sometimes undefined.

Reply:
It depends on how we do that tbh. I mentioned somewhere that we might want to do a "frozen solvent component" where we tell edges they absolutely cannot change anything about the solvent.

Also does it matter if the edge does something else than what the components on the node currently look like? Isn't the whole point of this method so that users & edges know what the node components look like and then use that to do further things? (parameterisation, charge corrections, etc...)

Reply:
I think we can pursue a policy where edge protocols don't do anything in processing that would contradict the node definition. This means that if e.g. box vectors are fully specified in two nodes, an edge protocol *must* use those, and if it *can't* then it must fail upon being defined. 

In other words, node properties define a policy, and protocols should produce systems to be computed in line with the policies of the nodes they connect. If the policies are too strict for the protocol to work with, it should throw up its hands immediately on definition; if the policies are loose enough, it has that much ground to work with in downstream processing to produce computable MD systems.

Does this make sense?

Reply:
Yep that's a good definition.